### PR TITLE
Align CMSSW_7_4_X to CMSSW_7_4_THREADED_X.

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

**Comment by civanch** [comment]
+1

File: RecoEgamma/ElectronIdentification/plugins/ElectronIdMVABased.cc [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
 using namespace std;
 using namespace reco;
-class ElectronIdMVABased : public edm::EDFilter {
+class ElectronIdMVABased : public edm::stream::EDFilter {
```
Comment (slava77) [inline RecoEgamma/ElectronIdentification/plugins/ElectronIdMVABased.cc]: ``` EDFilter<>```

**Comment by slava77** [comment]
-1

simple update is needed to compile

**Comment by ktf** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
@degano, @nclopezo : Jenkins tests were triggered 16 hours ago, but there are still no results. Could you please check what might be wrong?

Thanks.

**Comment by nclopezo** [comment]
Hi, It seems that there was an issue with the jenkins node that was going to execute the tests. I restarted them.

**Comment by civanch** [comment]
+1

**Comment by cvuosalo** [comment]
The static analyzer complains about  RecoJets/JetProducers/plugins/VirtualJetProducer.h, a file being changed in this PR. There's a static data member "names" that could be thread unsafe. If this PR is intended to make changes to enable multi-threading, it would be nice to eliminate this warning my defining "names" as:

static const char *const names[];

and changing the initialization of "names" in VirtualJetProducer.cc accordingly.

**Comment by cvuosalo** [comment]
+1

For #7817 22f548694f8a8dc35f040a9a64735a1d6ca36ab9

Very small code changes to facilitate multi-threading, which are not expected to cause any changes in reconstruction.

Code changes are satisfactory. Jenkins tests against [CMSSW_7_4_X_2015-02-22-0200] (https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7817/2826/summary.html) show no significant differences.

**Pull request merged**